app/models/concerns/obstructed.rb

```diff
@@ -0,0 +1,70 @@
+# In queen.rb, rook.rb, bishop.rb, pawn.rb, I added "include Obstructed"
+# Every time one of these pieces is moved the "valid_move?" method should be called
+# and that should call the "is_obstructed?" method.
+# game.pieces.where(horizontal_position & vertical_position) are extracted current location of pieces on game board.
+# If is_obstructed returns false, move is legal.
+# x_start, x_end, y_start, y_end values are filled when piece is moved.
+# Will write tests later after moving pieces action is more clear...
+module Obstructed
+  def horizontal_check

```

Comment:
For each of the check methods, we need to pass in x_end and y_end to make them available to the method.